client/lib/resize-image-url/index.js

```diff
@@ -39,20 +54,43 @@ const scaleByFactor = ( value ) => value * IMAGE_SCALE_FACTOR;
  * Changes the sizing parameters on a URL. Works for WordPress.com, Photon, and
  * Gravatar images
  *
- * @param   {String} imageUrl Original image url
- * @param   {Object} params   Resize parameters to add
- * @returns {String}          Resize image URL
+ * @param   {String}          imageUrl Original image url
+ * @param   {(Number|Object)} resize   Resize pixel width, or object of query
+ *                                     arguments (assuming Photon or Gravatar)
+ * @param   {?Number}         height   Pixel height if specifying resize width
+ * @returns {String}                   Resize image URL
  */
-export default function resizeImageUrl( imageUrl, params ) {
+export default function resizeImageUrl( imageUrl, resize, height ) {
+	if ( 'string' !== typeof imageUrl ) {
+		return imageUrl;
+	}
+
 	const parsedUrl = parse( imageUrl, true, true );
 	if ( ! REGEXP_VALID_PROTOCOL.test( parsedUrl.protocol ) ) {
 		return imageUrl;
 	}
 
 	parsedUrl.query = omit( parsedUrl.query, SIZE_PARAMS );
 
+	if ( 'number' === typeof resize ) {
```

Comment:
In my ideal world, the function should have this signature: `resizeImageUrl( url, config )` and `config` here could have some documented attributes and these attributes should be meaningful no matter the url we provide (Gravatar, wp, photon or external).

```js
resizeImageUrl( url, { width: 50 } );
resizeImageUrl( url, { width: 50, height: 100 } );
resizeImageUrl( url, { fit } );
```

I don't mind calling it with a { width } instead of using width directly as a param. But I understand that you could find some value in simpler signatures `url, width, height` and yes the documentation helps in this case and the use cases you provided are clear enough :).